harness/determined/common/util.py

```diff
@@ -104,6 +104,43 @@ def wrapped(*arg: Any, **kwarg: Any) -> Any:
     return wrapped
 
 
+def yaml_safe_dump(
+    data: Any,
+    stream: Optional[Union[io.FileIO, IO[Any], str]] = None,
+    default_flow_style: Optional[bool] = None,
+) -> Any:
+    """
+    A utility wrapper to mimick the pre-0.18.0 ruamel.yaml.safe_dump() API.
+
+    The new API is needlessly verbose, and has some benefits we don't care about.
+    """
+
+    if default_flow_style is None:
+        # Reuse-existing YAML object.
+        y = _yaml
+    else:
+        y = yaml.YAML(typ="safe", pure=True)
+        y.default_flow_style = default_flow_style
```

Comment:
not sure if there's much performance impact here, but maybe just make second instance next to `_yaml` which has `default_flow_style=False`?

Reply:
sure

Reply:
what I actually did was create the yaml objects lazily and cache them in a dict.  I figure if we think they're expensive enough to not want to create two or three of them, then they're also expensive enough we probably don't want to create one of them that won't be used.